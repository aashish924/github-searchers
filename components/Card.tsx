import React, { ReactNode } from 'react'
import styled from 'styled-components';


const CardWrapper = styled.div`
  display: flex;
  justify-content: center;
  margin 16px 8px;
  width: 300px;
  height: 232px;
  border: 1px solid gray;
`;

const CardHeader = styled.div`
  background-color: #f5f5f5;
  height: auto;
  padding: 8px;
  border-bottom: 1px solid gray;
`;


interface SearchResultProps {
    children?: ReactNode;
    style?: any;
    title?: ReactNode|string;
}


const Card = (props: SearchResultProps) => {

  return <CardWrapper style={props.style}>
    {props.title ? <CardHeader>{props.title}</CardHeader> : ""}
    {props.children}
  </CardWrapper>
};

export default Card;