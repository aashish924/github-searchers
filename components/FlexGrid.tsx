import React, { ReactNode } from 'react'
import styled from 'styled-components';


const MainContainer = styled.div`
`;

const WrapperContainer = styled.div`
  width: 100%;
  height: 100%;
  margin: 8px;
  display: flex;
  flex-wrap: wrap;
`;


interface SearchResultProps {
    children?: ReactNode;
}


const FlexGrid = (props: SearchResultProps) => {

  return <MainContainer>
    <WrapperContainer>
      {props.children}
    </WrapperContainer>
  </MainContainer>
};

export default FlexGrid;