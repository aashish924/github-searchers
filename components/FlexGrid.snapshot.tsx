import * as React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card'
import FlexGrid from './FlexGrid'


describe('FlexGrid should render correctly', () => {
  it('Empty FlexGrid', () => {
    const tree = renderer.create(<FlexGrid />).toJSON();
    expect(tree).toMatchSnapshot('empty-flexgtid');
  });

  it('FlexGrid with some cards', () => {
    const cards = <FlexGrid>
      <Card title="First Card" />
      <Card title="Second Card" />
      <Card title="Third Card">
        <h3>Hello World!</h3>
      </Card>
    </FlexGrid>;
    const tree = renderer.create(cards).toJSON();
    expect(tree).toMatchSnapshot('non-empty-flexgrid');
  });
});