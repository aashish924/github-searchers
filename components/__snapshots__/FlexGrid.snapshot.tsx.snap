// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexGrid should render correctly Empty FlexGrid: empty-flexgtid 1`] = `
<div
  className="sc-AxiKw cydMkg"
>
  <div
    className="sc-AxhCb ctdKxn"
  />
</div>
`;

exports[`FlexGrid should render correctly FlexGrid with some cards: non-empty-flexgrid 1`] = `
<div
  className="sc-AxiKw cydMkg"
>
  <div
    className="sc-AxhCb ctdKxn"
  >
    <div
      className="sc-AxjAm iEZjyM"
    >
      <div
        className="sc-AxirZ kUTnFM"
      >
        First Card
      </div>
    </div>
    <div
      className="sc-AxjAm iEZjyM"
    >
      <div
        className="sc-AxirZ kUTnFM"
      >
        Second Card
      </div>
    </div>
    <div
      className="sc-AxjAm iEZjyM"
    >
      <div
        className="sc-AxirZ kUTnFM"
      >
        Third Card
      </div>
      <h3>
        Hello World!
      </h3>
    </div>
  </div>
</div>
`;
